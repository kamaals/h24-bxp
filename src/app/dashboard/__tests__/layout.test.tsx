import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import DashboardLayout from "../layout";
import { vi } from "vitest";
import ReduxStoreProvider from "@/components/providers/redux-store-provider";

vi.mock("next/navigation", () => ({
  useRouter: vi.fn(),
  usePathname: vi.fn(),
}));

vi.mock("@/lib/store/hooks", () => ({
  useAppSelector: vi.fn(),
}));

describe("Dashboard Layout ", () => {
  it("Should match the snapshot", () => {
    const result = render(
      <ReduxStoreProvider user={null}>
        <DashboardLayout breadcrumb={<div>Breadcrumb</div>}>
          <div>Test</div>
        </DashboardLayout>
      </ReduxStoreProvider>,
    );
    expect(result).toMatchSnapshot();
  });
});
