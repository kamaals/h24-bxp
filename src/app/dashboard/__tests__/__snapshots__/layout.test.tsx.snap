// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dashboard Layout  > Should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="group/sidebar-wrapper has-data-[variant=inset]:bg-sidebar flex min-h-svh w-full"
        data-slot="sidebar-wrapper"
        style="--sidebar-width: 16rem; --sidebar-width-icon: 3rem;"
      >
        <main
          class="bg-background relative flex w-full flex-1 flex-col md:peer-data-[variant=inset]:m-2 md:peer-data-[variant=inset]:ml-0 md:peer-data-[variant=inset]:rounded-xl md:peer-data-[variant=inset]:shadow-sm md:peer-data-[variant=inset]:peer-data-[state=collapsed]:ml-2"
          data-slot="sidebar-inset"
        >
          <header
            class="bg-sidebar w-full z-30 fixed flex h-16 shrink-0 items-center gap-2 transition-[width,height] ease-linear group-has-[[data-collapsible=icon]]/sidebar-wrapper:h-12"
          >
            <div
              class="flex items-center gap-2 px-4"
            >
              <button
                class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive hover:bg-accent hover:text-accent-foreground dark:hover:bg-accent/50 size-7 -ml-1"
                data-sidebar="trigger"
                data-slot="sidebar-trigger"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-panel-left"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    height="18"
                    rx="2"
                    width="18"
                    x="3"
                    y="3"
                  />
                  <path
                    d="M9 3v18"
                  />
                </svg>
                <span
                  class="sr-only"
                >
                  Toggle Sidebar
                </span>
              </button>
              <div
                class="bg-border shrink-0 data-[orientation=horizontal]:h-px data-[orientation=horizontal]:w-full data-[orientation=vertical]:h-full data-[orientation=vertical]:w-px mr-2 h-4"
                data-orientation="vertical"
                data-slot="separator-root"
                role="none"
              />
              <div>
                Breadcrumb
              </div>
            </div>
          </header>
          <section
            class="p-6 pt-24"
          >
            <div>
              Test
            </div>
          </section>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="group/sidebar-wrapper has-data-[variant=inset]:bg-sidebar flex min-h-svh w-full"
      data-slot="sidebar-wrapper"
      style="--sidebar-width: 16rem; --sidebar-width-icon: 3rem;"
    >
      <main
        class="bg-background relative flex w-full flex-1 flex-col md:peer-data-[variant=inset]:m-2 md:peer-data-[variant=inset]:ml-0 md:peer-data-[variant=inset]:rounded-xl md:peer-data-[variant=inset]:shadow-sm md:peer-data-[variant=inset]:peer-data-[state=collapsed]:ml-2"
        data-slot="sidebar-inset"
      >
        <header
          class="bg-sidebar w-full z-30 fixed flex h-16 shrink-0 items-center gap-2 transition-[width,height] ease-linear group-has-[[data-collapsible=icon]]/sidebar-wrapper:h-12"
        >
          <div
            class="flex items-center gap-2 px-4"
          >
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive hover:bg-accent hover:text-accent-foreground dark:hover:bg-accent/50 size-7 -ml-1"
              data-sidebar="trigger"
              data-slot="sidebar-trigger"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-panel-left"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="18"
                  rx="2"
                  width="18"
                  x="3"
                  y="3"
                />
                <path
                  d="M9 3v18"
                />
              </svg>
              <span
                class="sr-only"
              >
                Toggle Sidebar
              </span>
            </button>
            <div
              class="bg-border shrink-0 data-[orientation=horizontal]:h-px data-[orientation=horizontal]:w-full data-[orientation=vertical]:h-full data-[orientation=vertical]:w-px mr-2 h-4"
              data-orientation="vertical"
              data-slot="separator-root"
              role="none"
            />
            <div>
              Breadcrumb
            </div>
          </div>
        </header>
        <section
          class="p-6 pt-24"
        >
          <div>
            Test
          </div>
        </section>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
